### PR TITLE
Support for "Modified Files Coverage"

### PR DESCRIPTION
**Pull request by fo-code**
I added support for "Modified Files Coverage".
Also, "Change Coverage" is renamed to "Modified Lines Coverage" to separate the coverage metrics more easily.

### CODE REVIEW

File: src/main/java/edu/hm/hafner/metric/FileNode.java [hunk 1]
```diff
@@ -119,7 +120,24 @@ else if (covered + missed == 1) {
...
+        if (hasCoveredLinesInChangeSet()) {
+            var copy = copy();
+            var lineCoverage = Coverage.nullObject(Metric.LINE);
+            var branchCoverage = Coverage.nullObject(Metric.BRANCH);
+            for (int line : getCoveredLines()) {
+                lineCoverage = lineCoverage.add(getLineCoverage(line));
+                branchCoverage = branchCoverage.add(getBranchCoverage(line));
+            }
+            addLineAndBranchCoverage(copy, lineCoverage, branchCoverage);
+            return Optional.of(copy);
```
Comment (uhafner) [inline src/main/java/edu/hm/hafner/metric/FileNode.java:134]: Can't we simply copy the file with all children and values?

File: src/main/java/edu/hm/hafner/metric/FileNode.java [hunk 2]
```diff
@@ -238,7 +256,7 @@ public Coverage getBranchCoverage(final int line) {
             var covered = getCoveredOfLine(line);
             var missed = getMissedOfLine(line);
             if (covered + missed > 1) {
-                return new CoverageBuilder().setMetric(Metric.LINE)
+                return new CoverageBuilder().setMetric(Metric.BRANCH)
```
Comment (uhafner) [inline src/main/java/edu/hm/hafner/metric/FileNode.java:259]: Good spot!

File: src/main/java/edu/hm/hafner/metric/Node.java [hunk 3]
```diff
@@ -733,6 +733,32 @@ protected Optional<Node> filterByChanges() {
         return Optional.of(copy);
     }
 
+    /**
+     * Creates a new coverage tree that represents the Changed Files Coverage. This new tree will contain only those
```
Comment (uhafner) [inline src/main/java/edu/hm/hafner/metric/Node.java:737]: Maybe we should use modified lines and modified files everywhere?

File: src/main/java/edu/hm/hafner/metric/Node.java [hunk 4]
```diff
@@ -733,6 +733,32 @@ protected Optional<Node> filterByChanges() {
...
+                .stream()
+                .map(Node::filterByChangedFiles)
+                .flatMap(Optional::stream)
+                .collect(Collectors.toList());
+        if (prunedChildren.isEmpty()) {
+            return Optional.empty();
+        }
+        copy.addAllChildren(prunedChildren);
+        return Optional.of(copy);
+    }
```
Comment (uhafner) [inline src/main/java/edu/hm/hafner/metric/Node.java:760]: It makes sense to extract that method into a private method that gets the mapping `Function` as an argument (I should have done this already in my code 🙈 )

**Review by uhafner — CHANGES_REQUESTED** [review]

File: src/main/java/edu/hm/hafner/metric/FileNode.java (hunk 1 above)
Comment (fo-code) [reply src/main/java/edu/hm/hafner/metric/FileNode.java:134]: You are right!

File: src/main/java/edu/hm/hafner/metric/Node.java (hunk 3 above)
Comment (fo-code) [reply src/main/java/edu/hm/hafner/metric/Node.java:737]: Yes, that sounds goot to me.

File: src/main/java/edu/hm/hafner/metric/Node.java (hunk 4 above)
Comment (fo-code) [reply src/main/java/edu/hm/hafner/metric/Node.java:760]: Yes, true!

**Review by uhafner — APPROVED** [review]

File: src/main/java/edu/hm/hafner/metric/Metric.java [hunk 5]
```diff
@@ -74,6 +79,20 @@ public static boolean isNodeMetric(final Metric metric) {
...
+                Metric.CONTAINER,
+                Metric.MODULE,
+                Metric.PACKAGE,
+                Metric.FILE,
+                Metric.CLASS,
+                Metric.METHOD,
+                Metric.LINE,
+                Metric.BRANCH,
+                Metric.INSTRUCTION
+        ));
```
Comment (uhafner) [inline src/main/java/edu/hm/hafner/metric/Metric.java:93]: Where do you need this method?

**Pull request merged**